### ISSUE TITLE
Blog: author profile links

### ISSUE THREAD

**Issue by aborruso** [squidfunk/mkdocs-material#5955]
### Description

Hi,
I think it could be useful to have the option to associate a reference url to each author (this could be the author website, the author mastodon URL, the authot github profile, ecc., what the author prefers).

Something like this:

```
authors:
  squidfunk:
    name: Martin Donath
    description: Creator
    avatar: https://github.com/squidfunk.png
    reference_url: https://fosstodon.org/@squidfunk
```

In this way, in example, if the author has a reference URL, we could have an hypelink associate to author  name, in the single page blog post template (see image example below).

![image](https://github.com/squidfunk/mkdocs-material/assets/30607/2b600825-05f4-4c21-aad3-25d8a645f2d2)

### Related links

- https://github.com/squidfunk/mkdocs-material/discussions/5953

### Use Cases

In this way each author can make himself/herself better known.

### Before submitting

- [X] I have read and followed the [change request guidelines](https://squidfunk.github.io/mkdocs-material/contributing/requesting-a-change/).
- [X] I have verified that [my idea is a change request and not a bug report](https://squidfunk.github.io/mkdocs-material/contributing/requesting-a-change/#its-not-a-bug-its-a-feature).
- [X] I have ensured that, to the best of my knowledge, [my idea will benefit the entire community](https://squidfunk.github.io/mkdocs-material/contributing/requesting-a-change/#benefit-for-the-community).
- [X] I have included relevant links to [the documentation](https://squidfunk.github.io/mkdocs-material), related [issues](https://github.com/squidfunk/mkdocs-material/issues), and [discussions](https://github.com/squidfunk/mkdocs-material/discussions) to underline the need for my idea.

**Comment by alexvoss** [comment]
@squidfunk This is something I might be interested to have a go at. It sounds simple enough (famous last words). Just checking to see if there is a reason not to and to avoid possible duplication of work. Assign to me if happy.

**Comment by squidfunk** [comment]
Thanks for tackling this change request! However, please see https://github.com/squidfunk/mkdocs-material/pull/6038#issuecomment-1724905838 – we'd like to think a little bigger here ☺️

**Comment by bigpick** [comment]
> However, we already have bigger plans for this, i.e., allowing authors to generate profile pages that list all of their articles, similar to category and archive pages, and provide arbitrary additional information. Thus, we're currently not considering merging this PR, but  
it's definitely helpful as a customization for those wishing for this functionality right now.

... is there a ticket/issue/something to watch for progress on that? Or is this the correct place? Thanks!

**Comment by squidfunk** [comment]
This is the right place. You can subscribe to this issue (which you should be now, since you commented).

**Comment by squidfunk** [comment]
I've added this functionality to `master` in 28ebcbf (tagged you as a co-author, @alexvoss). I've slightly adjusted #6038. You can now define the `url` property for an author in `.authors.yml`, and the micro-profile will automatically add a link.

Additionally, Insiders brings a new feature: __author profiles__ in [`a9ce9d2`](https://github.com/squidfunk/mkdocs-material-insiders/commit/a9ce9d204ac5f8cfe61f7a3a1394e1523dd36687)! The blog plugin can now create a custom page for each author that lists all of the author's blog posts in reverse chronological order. Additionally, when combining this with [custom index pages](https://squidfunk.github.io/mkdocs-material/setup/setting-up-a-blog/#custom-index-pages), you can create a custom profile for an author, preceding the list of blog posts written by the author. We'll soon add examples for this (cc @alexvoss). The following new settings were added:

``` yaml
plugins:
  - blog:
      authors_profiles: true
      authors_profiles_name: Authors
      authors_profiles_url_format: "author/{name}"
      authors_profiles_pagination: true
      authors_profiles_pagination_per_page: 10
      authors_profiles_toc: false
```

If you omit the `url` in `.authors.yml` and enable the `author_profiles` setting, the micro-profile will automatically link to the automatically generated author profile. This yields the opportunity to provide much more background information, social links, etc., for an author than you could squeeze into the micro-profile, and still have it linked from each and every post.

I'll update this comment once I finished writing the docs for this, but I wanted to let you know that this is now available.

**Comment by aborruso** [comment]
THank you very much @squidfunk

**Comment by squidfunk** [comment]
Keeping open until released, as always.

**Comment by aborruso** [comment]
> Keeping open until released, as always.

I'm very sorry. Thank you again

**Comment by squidfunk** [comment]
Released as part of [9.4.14](https://squidfunk.github.io/mkdocs-material/changelog/#9.4.14). Additionally, author profiles have been released as part of [9.4.14+insiders-4.46.0](https://squidfunk.github.io/mkdocs-material/insiders/changelog/#4.46.0).
